docs/user-guide/creating-application/docker-build-configuration.md

```diff
@@ -1,59 +1,105 @@
 
 **If you are using docker hub account, you need to enter the repository name along with your username. For example - If my username is *kartik579* and repo name is *devtron-trial*, then enter kartik579/devtron-trial instead of only devtron-trial.**
 
 ![](../../.gitbook/assets/docker-configuration-docker-hub.png)
 
-![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/docker-build-configuration/docker-build-config-container-registry1.png)
 
+## Build the Container Image
+
+In order to deploy the application, we must build the container images to configure a fully operational container environment.
+
+You can choose one of the following options to build your container image:
+* **I have a Dockerfile**
+* **Create Dockerfile**
+* **Build without Dockerfile**
+
+### Build Docker Image when you have a Dockerfile
+
+A `Dockerfile` is a file that you create which in turn produces a Docker image when you build it.
+
+![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/docker-build-configuration/i-have-a-dockerfile.png)
+
+| Field | Description |
+| --- | --- |
+| **Select repository containing Dockerfile** | Select the Git checkout path of your repository. This repository is the same which you defined on the [Git Repository](https://docs.devtron.ai/usage/applications/creating-application/git-material) section. |
+| **Dockerfile Path (Relative)** | Enter a relative file path where your docker file is located in Git repository. Ensure that the dockerfile is available on this path. This is a mandatory field. |
+
+### Build Docker Image by creating Dockerfile
+
+With the option **Create Dockerfile**, you can create a `Dockerfile` from the available templates. You can edit any selected Dockerfile template as per your build configuration requirements.
+
+![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/docker-build-configuration/create-dockerfile.jpg)
+
+| Field | Description |
+| --- | --- |
+| **Language** | Select the programming language (e.g., `Java`, `Go`, `Python`, `Node` etc.) from the drop-down list you want to create a dockerfile as per compatibility to your system.<br>**Note** We will be adding other programming languages in the future releases.</br>|
+| **Framework** | Select the framework (e.g., `Maven`, `Gradle` etc.) of the selected programming language.<br>**Note** We will be adding other frameworks in the future releases.</br>|
+
+### Build Docker Image without Dockerfile
 
-## Checkout path 
-Checkout path including inputs:
-1. Git checkout path
-2. Docker file (relative)
+With the option **Build without Dockerfile**, you can use buildpacks to automate the build processes for your preferred languages and frameworks.
```

Comment:
With the option **Build without Dockerfile**, you can use Buildpacks to automatically build the image for your preferred language and framework.